image/bmp.cpp

```diff
@@ -96,12 +96,18 @@ bool BitmapDecoder::loadStream(Common::SeekableReadStream &stream) {
 	/* uint16 planes = */ stream.readUint16LE();
 	uint16 bitsPerPixel = stream.readUint16LE();
 
-	if (bitsPerPixel != 4 && bitsPerPixel != 8 && bitsPerPixel != 24 && bitsPerPixel != 32) {
+	if (bitsPerPixel != 4 && bitsPerPixel != 8 && bitsPerPixel != 16 && bitsPerPixel != 24 && bitsPerPixel != 32) {
 		warning("%dbpp bitmaps not supported", bitsPerPixel);
 		return false;
 	}
 
 	uint32 compression = stream.readUint32BE();
+
+	if (bitsPerPixel == 16 && compression == 3) {
```

Comment:
To things here:
First, the compression is read BE so constant should be reversed like it is in `createBitmapCodec` using `SWAP_CONSTANT_32`.
Second, as RLE codec doesn't support 16-bits bitmaps, it would be safer to check that value is really `SWAP_CONSTANT_32(0)`.

Reply:
Thanks! Changed